src/core/Microsoft.PowerToys.Settings.UI.Runner/MainWindow.xaml.cs

```diff
@@ -70,17 +69,20 @@ private void WindowsXamlHost_ChildChanged(object sender, EventArgs e)
                 {
                     if (ShellPage.ShellHandler.IPCResponseHandleList != null)
                     {
-                        try
+                        var success = JsonObject.TryParse(msg, out JsonObject json);
+                        if (success)
                         {
-                            JsonObject json = JsonObject.Parse(msg);
                             foreach (Action<JsonObject> handle in ShellPage.ShellHandler.IPCResponseHandleList)
                             {
                                 handle(json);
                             }
                         }
-                        catch (Exception e)
+                        else
                         {
-                            Logger.LogError("Exception encountered while handling IPC response", e);
+                            Logger.LogError("Failed to parse JSON from IPC message.");
+#if DEBUG
```

Comment:
Sorry for the confusion Luthfi.  I would put this inside the LogError method for consistency.